src/town_cmd.cpp

```diff
@@ -2879,15 +2883,26 @@ static bool TryBuildTownHouse(Town *t, TileIndex tile)
 		/* Special houses that there can be only one of. */
 		t->flags |= oneof;
 
-		BuildTownHouse(t, tile, hs, house, random_bits);
+		/* Should the house be protected? */
+		bool protect = hs->extra_flags.Test(HouseExtraFlag::BuildingIsProtected);
+
+		BuildTownHouse(t, tile, hs, house, random_bits, protect);
 
 		return true;
 	}
 
 	return false;
 }
 
-CommandCost CmdPlaceHouse(DoCommandFlag flags, TileIndex tile, HouseID house)
+/**
+ * Place a house manually.
+ * @param flags Type of operation.
+ * @param tile Tile on which to place the house.
+ * @param HouseID The HouseID of the house spec.
+ * @param protect Should the house be protected?
```

Comment:
```suggestion
 * @param protect Whether the house be protected
```

Having parameters with a question-mark are always a bit awkward.